### PR TITLE
wlr-layout-ui: init at 1.4.7

### PR DESCRIPTION
**Pull request by bnlrnz**
## Description of changes

Adding a Python-based package that provides a UI to configure monitor layouts for wlr based systems.
[Link to the original Repo](https://github.com/fdev31/wlr-layout-ui/)

This PR is related to Issue https://github.com/NixOS/nixpkgs/issues/298342

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/wl/wlr-layout-ui/package.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    tomli
+    tomli-w
+  ];
+
+  meta = with lib; {
+    description = "A simple GUI to setup the screens layout on wlroots based systems.";
+    homepage = "https://github.com/fdev31/wlr-layout-ui/";
+    changelog = "https://github.com/fdev31/wlr-layout-ui/releases/tag/${src.rev}";
+    maintainers = with maintainers; [ bnlrnz ];
+    license = licenses.unfree;
```
Comment (Aleksanaa) [inline pkgs/by-name/wl/wlr-layout-ui/package.nix:33]: I'd suggest you open an issue and ask the author which license they tend to use. L

File: pkgs/by-name/wl/wlr-layout-ui/package.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  ];
+
+  propagatedBuildInputs = with python3.pkgs; [
+    pyglet
+    tomli
+    tomli-w
+  ];
+
+  meta = with lib; {
+    description = "A simple GUI to setup the screens layout on wlroots based systems.";
```
Comment (Aleksanaa) [inline pkgs/by-name/wl/wlr-layout-ui/package.nix:29]:
```suggestion
    description = "A simple GUI to setup the screens layout on wlroots based systems";
```

File: pkgs/by-name/wl/wlr-layout-ui/package.nix [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    homepage = "https://github.com/fdev31/wlr-layout-ui/";
+    changelog = "https://github.com/fdev31/wlr-layout-ui/releases/tag/${src.rev}";
```
Comment (Aleksanaa) [inline pkgs/by-name/wl/wlr-layout-ui/package.nix:31]: The upstream author does not seem to have released a release or written a changelog. We should not add this changelog in this case.

File: pkgs/by-name/wl/wlr-layout-ui/package.nix (hunk 1 above)
Comment (Aleksanaa) [reply pkgs/by-name/wl/wlr-layout-ui/package.nix:33]: Also ask the author. If the number of contributors is limited, they can decide how to distribute legacy code (without violating the licenses of others).

File: pkgs/by-name/wl/wlr-layout-ui/package.nix [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    tomli
+    tomli-w
+  ];
+
+  meta = with lib; {
+    description = "A simple GUI to setup the screens layout on wlroots based systems";
+    homepage = "https://github.com/fdev31/wlr-layout-ui/";
+    maintainers = with maintainers; [ bnlrnz ];
+    license = licenses.mit;
+    mainProgram = "wlrlui";
```
Comment (Aleksanaa) [inline pkgs/by-name/wl/wlr-layout-ui/package.nix:33]:
```suggestion
    mainProgram = "wlrlui";
    platforms = subtractLists platforms.darwin platforms.unix;
```

As building on darwin is useless

**Review by Aleksanaa — APPROVED** [review]
Please squash your commits.

**Comment by Aleksanaa** [comment]
Doesn't work:

```
=> ./result/bin/wlrlui 
Traceback (most recent call last):
  File "/nix/store/ablv4jas0wnyaya6qdcy00a11pjf07zp-wlr-layout-ui-1.4.3/bin/.wlrlui-wrapped", line 9, in <module>
    sys.exit(main())
             ^^^^^^
  File "/nix/store/ablv4jas0wnyaya6qdcy00a11pjf07zp-wlr-layout-ui-1.4.3/lib/python3.11/site-packages/wlr_layout_ui/__init__.py", line 56, in main
    load()
  File "/nix/store/ablv4jas0wnyaya6qdcy00a11pjf07zp-wlr-layout-ui-1.4.3/lib/python3.11/site-packages/wlr_layout_ui/screens.py", line 92, in load
    load_from_hyprctl()
  File "/nix/store/ablv4jas0wnyaya6qdcy00a11pjf07zp-wlr-layout-ui-1.4.3/lib/python3.11/site-packages/wlr_layout_ui/screens.py", line 53, in load_from_hyprctl
    modes = [Mode(*_parseMode(m)) for m in monitor["availableModes"]]
                                           ~~~~~~~^^^^^^^^^^^^^^^^^^
KeyError: 'availableModes'
```

Also you've not installed their desktop file (or create one yourself) https://github.com/fdev31/wlr-layout-ui/blob/main/files/wlr-layout-ui.desktop

**Review by Aleksanaa — COMMENTED** [review]
Also does this work for you? I can't test, possibly because of hyprland

File: pkgs/by-name/wl/wlr-layout-ui/package.nix [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+  propagatedBuildInputs = with python3.pkgs; [
+    pyglet
+    tomli
+    tomli-w
+  ];
+
+  postInstall = ''
+    mkdir -p $out/share/applications
+    cp files/wlr-layout-ui.desktop $out/share/applications
```
[the remark this answers is not in the document]
Comment (bnlrnz) [reply pkgs/by-name/wl/wlr-layout-ui/package.nix]: The default.nix file of hyprland changes the version command behaviour of hyprctl. I patched a file in wlr-layout-ui to catch this change.

**Review by Aleksanaa — APPROVED** [review]
Sort of works

**Pull request merged**